torch/testing/_internal/common_methods_invocations.py

```diff
@@ -349,6 +349,70 @@ def wrapped_fn(x):
 
     return wrapped_fn
 
+def np_binary_ufunc_type_promotion_wrapper(fn):
+    # Wrapper that passes PyTorch's default scalar
+    #   type as an argument to the wrapped NumPy
+    #   binary ufunc when given an integer input .
+    #   This mimicks PyTorch's integer->floating point
+    #   type promotion.
+    #
+    # This is necessary when NumPy promotes
+    #   integer types to double, since PyTorch promotes
+    #   integer types to the default scalar type.
+    #
+    # Besides, the table below shows the difference between
+    #   PyTorch and NumPy on the type promotion for binary
+    #   functions
+    #      inputs         NumPy     PyTorch
+    #    bool  bool       vary        bool
+    #   int16 float16    float32    float16
+    #   int32 float16    float64    float16
+    #   int64 float16    float64    float16
+    #   int32 float32    float64    float32
+    #   int64 float32    float64    float32
+    #
+    # Another special case only on Windows platform due to
+    # torch tensor wrapper for integer scalar is int64, but
+    # numpy wrapper for integer scalar is int32.
+    # For boolean tensor and integer scalar, NumPy result
+    # needs to be promoted from int32 to int64
+
+    # Helper to get the unified type between PyTorch
+    #    and NumPy
+    #    x is Tensor; y can be Tensor or Scalar
+    def unified_type(x, y):
```

Comment:
nit: if you are only returning dtypes, probably want to have "dtype" in the name rather than "type"

Reply:
what is the definition of a unified type?  Is the big comment above about `np_binary_ufunc_type_promotion_wrapper` and not `unified_type`?

Reply:
Good point, I should make it clear. `np_binary_ufunc_type_promotion_wrapper` is a wrapper to generate a new numpy function with the corresponding dtype for the scenarios shown in the comment.